scrapy_poet/page_input_providers.py

```diff
@@ -117,7 +142,17 @@ class ResponseDataProvider(PageObjectInputProvider):
     """This class provides ``web_poet.page_inputs.ResponseData`` instances."""
 
     provided_classes = {ResponseData}
+    name = "response_data"
 
     def __call__(self, to_provide: Set[Callable], response: Response):
         """Builds a ``ResponseData`` instance using a Scrapy ``Response``"""
         return [ResponseData(url=response.url, html=response.text)]
+
+    def fingerprint(self, to_provide: Set[Callable], request: Request) -> str:
+        return request_fingerprint(request)
```

Comment:
When would a developer prefer built-in Scrapy's cache, and when should ResponseDataProvider's cache be used?

Reply:
Also, about fingerprints. The approach here makes total sense, as this is what built-in Scrapy cache is using - request_fingerprint. It also has a performance advantage, as these fingerprints are used in many places, and they're cached. 

However, these fingerprints are stored in sqlite columns now. So, when deciding on a fingerprint for AutoExtract provider, I opted in json-encoded request data (https://github.com/scrapinghub/scrapy-autoextract/blob/85f6318a9212f53300ad9569248600e25e98017e/scrapy_autoextract/cache.py#L72). It is much less efficient, and takes disk space, but it's much easier to inspect the cache - e.g. find and delete cache records for a certain domain, or a certain URL.

Reply:
> When would a developer prefer built-in Scrapy's cache, and when should ResponseDataProvider's cache be used?

I shall be adding this in the docs in a succeeding commit.

> It is much less efficient, and takes disk space, but it's much easier to inspect the cache - e.g. find and delete cache records for a certain domain, or a certain URL.

Those trade-offs makes sense to me. I also agree about prioritizing debuggability first and foremost. Should performance deterioration be observed on certain scenarios, the user could always override the `fingerprint()` method.

I've made changes in-line with this on https://github.com/scrapinghub/scrapy-poet/pull/55/commits/734bd5ed0abe45ba6b15b671b81d6a7d57acd216. I've still preserved Scrapy's Request Fingerprint so that the user can easily use it as a reference later on if needed.